smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/ProtocolGenerator.kt

```diff
@@ -42,21 +42,11 @@ interface ProtocolGenerator {
         fun getFormattedDateString(
             tsFormat: TimestampFormatTrait.Format,
             memberName: String,
-            isUnwrapped: Boolean = true,
-            roundEpoch: Boolean = false,
             urlEncode: Boolean = false
         ): String {
-            val terminator = if (isUnwrapped) "" else "?"
-            val epochTerminator = if (roundEpoch) ".clean" else ""
             val stringDateTerminator = if (urlEncode) ".urlPercentEncoding()" else ""
-            return when (tsFormat) {
-                TimestampFormatTrait.Format.EPOCH_SECONDS -> "${memberName}$terminator.timeIntervalSince1970$epochTerminator"
-                // FIXME return to this to figure out when to use fractional seconds precision in more general sense after we switch
-                // to custom date type
-                TimestampFormatTrait.Format.DATE_TIME -> "${memberName}$terminator.iso8601WithoutFractionalSeconds()$stringDateTerminator"
-                TimestampFormatTrait.Format.HTTP_DATE -> "${memberName}$terminator.rfc5322()$stringDateTerminator"
-                else -> throw CodegenException("unknown timestamp format: $tsFormat")
-            }
+            val timestampFormat = TimestampHelpers.generateTimestampFormatEnumValue(tsFormat)
+            return "TimestampFormatter(format: .$timestampFormat).string(from: $memberName)$stringDateTerminator"
```

Comment:
I don't have an obvious solution to convert this snippet to use CodeWriter for formatting. 
I'd like to keep this as is for now since it simply improves the existing implementation but lmk if you think this should be a blocker